google/cloud/internal/oauth2_external_account_credentials_test.cc

```diff
@@ -150,13 +152,14 @@ TEST(ExternalAccount, ParseAwsSuccess) {
       "https://sts.{region}.aws.example.com"
       "?Action=GetCallerIdentity&Version=2011-06-15";
   auto constexpr kTestTokenType = "urn:ietf:params:aws:token-type:aws4_request";
-  auto constexpr kTestAudience =
-      "//iam.googleapis.com/projects/$PROJECT_NUMBER/locations/global/"
-      "workloadIdentityPools/$POOL_ID/providers/$PROVIDER_ID";
+  auto const k_test_audience =
+      std::string{"//iam.googleapis.com/" +
+                  Location(Project("$PROJECT_NUMBER"), "global").FullName() +
```

Comment:
I am not sure this is a good application of the new class.  The URL path and resource names match in this case, but they don't have to match (e.g. storage uses`/b/$BUCKET/o/$OBJECT` for the URL path and `projects/_/buckets/$BUCKET/objects/$OBJECT` for the resource name).

Also, the leading `k_` is no longer applicable.

Reply:
Reverted.